controllers/deployment_controller.go

```diff
@@ -133,39 +133,44 @@ func (r *DeploymentReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 		return err
 	}
 
-	err = c.Watch(&source.Kind{Type: &csibaremetalv1.Deployment{}}, &handler.EnqueueRequestForObject{})
+	err = c.Watch(source.Kind(mgr.GetCache(), &csibaremetalv1.Deployment{}), &handler.EnqueueRequestForObject{})
```

Comment:
We didn't use caching before, is it necessary to use caching or something else？

Reply:
API change